boards/nordic/nrf54h20dk/Kconfig.defconfig

```diff
@@ -14,3 +14,11 @@ config BT_CTLR
 	default y if BT
 
 endif # BOARD_NRF54H20DK_NRF54H20_CPURAD
+
+if BOARD_NRF54H20DK_NRF54H20_CPUPPR
+
+# As PPR has limited memory most of tests does not fit with asserts enabled.
+config ASSERT
+	default n if ZTEST
```

Comment:
as most tests are actually smaller in footprint than actual samples, then why not simple disable `ASSERT` per default, and remove the `ZTEST` dependency (and perhaps even move it to the SoC implementation).

Especially as it will be clearer to a reader that `ASSERT` is `default n` in all cases for this board (or SoC CPU cluster if  moved).

Current construct might mislead the reader to think that `ASSERT` will by `y` when not `ZTEST`, which is actually not the case, ref:
https://github.com/zephyrproject-rtos/zephyr/blob/4d0a8c1e8b8316ec5071a3fcc98cd292d3bc4e6d/subsys/debug/Kconfig#L231-L233

Reply:
done